packages/cli/src/util/client.ts

```diff
@@ -132,7 +157,98 @@ export default class Client extends EventEmitter implements Stdio {
     });
   }
 
-  private _fetch(_url: string, opts: FetchOptions = {}) {
+  /**
+   * When the auth config is of type `OAuthAuthConfig`,
+   * this method silently tries to refresh the access_token if it is expired.
+   *
+   * If the refresh_token is also expired, it will not attempt to refresh it.
+   * If there is any error during the refresh process, it will not throw an error.
+   */
+  private async ensureAuthorized(): Promise<void> {
+    if (!isOAuthAuth(this.authConfig)) return;
+
+    const { authConfig } = this;
+
+    // If we have a valid access token, do nothing
+    if (isValidAccessToken(authConfig)) {
+      output.debug('Valid access token, skipping token refresh.');
+      return;
+    }
+
+    // If we don't have a valid refresh token, do nothing
+    if (!isValidRefreshToken(authConfig)) {
+      output.debug('Invalid refresh token, skipping token refresh.');
+      this.emptyAuthConfig();
+      this.writeToAuthConfigFile();
+      return;
+    }
+
+    const tokenResponse = await refreshTokenRequest({
+      refresh_token: authConfig.refreshToken,
+    });
+
+    const [tokensError, tokens] = await processTokenResponse(tokenResponse);
+
+    // If we had an error, during the refresh process, silently continue
+    if (tokensError) {
+      output.debug('Error refreshing token, skipping token refresh.');
+      if (output.isDebugEnabled()) output.prettyError(tokensError);
+      this.emptyAuthConfig();
+      this.writeToAuthConfigFile();
+      return;
+    }
+
+    this.updateAuthConfig({
+      type: 'oauth',
+      token: tokens.access_token,
+      expiresAt: Math.floor(Date.now() / 1000) + tokens.expires_in,
+    });
+
+    if (tokens.refresh_token) {
+      const [inspectError, payload] = await inspectToken(tokens.refresh_token);
+      if (inspectError) {
+        output.debug(
+          'Error verifying refresh token, skipping refresh token update.'
+        );
+        if (output.isDebugEnabled()) output.prettyError(inspectError);
+        this.emptyAuthConfig();
+        this.writeToAuthConfigFile();
+        return;
+      }
+
+      this.updateAuthConfig({
+        refreshToken: tokens.refresh_token,
+        refreshTokenExpiresAt: payload.exp,
+      });
```

Comment:
IMO, we could simplify this, and skip inspecting the refresh token:
```suggestion
      this.updateAuthConfig({
        refreshToken: tokens.refresh_token,
      });
```